doc/api/assert.markdown

```diff
@@ -56,47 +144,184 @@ is thrown instead.
+      // OK, 1 == '1'
+      
+    assert.equal(1, 2);
+      // AssertionError: 1 == 2
+    assert.equal({a: {b: 1}}, {a: {b: 1}});
+      //AssertionError: { a: { b: 1 } } == { a: { b: 1 } }
+
+If the values are not equal, an `AssertionError` is thrown with a `message`
+property set equal to the value of the `message` parameter. If the `message`
+parameter is undefined, a default error message is assigned.
 
 ## assert.fail(actual, expected, message, operator)
 
-Throws an `AssertionError`. If `message` is falsy, it displays the values for
-`actual` and `expected` separated by the provided `operator`. Otherwise, it
-displays `message` (and does not use `actual`, `expected`, and `operator`).
+Throws an `AssertionError`. If `message` is falsy, the error message is set as
+the values of `actual` and `expected` separated by the provided `operator`.
+Otherwise, the error message is the value of `message`.
+
+    const assert = require('assert');
+    
+    assert.fail(1, 2, undefined, '>');
+      // AssertionError: 1 > 2
+    
+    assert.fail(1, 2, 'whoops', '>');
+      // AssertionError: whoops
 
 ## assert.ifError(value)
 
 Throws `value` if `value` is truthy. This is useful when testing the `error`
 argument in callbacks.
 
+    const assert = require('assert');
+    
+    assert.ifError(0); // OK
+    assert.ifError(1); // Throws 1
+    assert.ifError('error') // Throws 'error'
+    assert.ifError(new Error()); // Throws Error
+
 ## assert.notDeepEqual(actual, expected[, message])
 
 Tests for any deep inequality. Opposite of [`assert.deepEqual`][].
 
+    const assert = require('assert');
+
+    const obj1 = {
+      a : {
+        b : 1
+      }
+    };
+    const obj2 = {
+      a : {
+        b : 2
+      }
+    };
+    const obj3 = {
+      a : {
+        b : 1
+      }
+    }
+    const obj4 = Object.create(obj1);
+
+    assert.deepEqual(obj1, obj1);
+      AssertionError: { a: { b: 1 } } notDeepEqual { a: { b: 1 } }
+      
+    assert.deepEqual(obj1, obj2);
+      // OK, obj1 and obj2 are not deep equal
+      
+    assert.deepEqual(obj1, obj3);
+      // AssertionError: { a: { b: 1 } } notDeepEqual { a: { b: 1 } }
+      
+    assert.deepEqual(obj1, obj4);
+      // OK, obj1 and obj2 are not deep equal
+
+If the values are deep equal, an `AssertionError` is thrown with a `message`
+property set equal to the value of the `message` parameter. If the `message`
+parameter is undefined, a default error message is assigned.
+
 ## assert.notDeepStrictEqual(actual, expected[, message])
 
-Tests for deep inequality. Opposite of [`assert.deepStrictEqual`][].
+Tests for deep strictly inequality. Opposite of [`assert.deepStrictEqual`][].
+
+    const assert = require('assert');
+
+    assert.notDeepEqual({a:1}, {a:'1'});
+      // AssertionError: { a: 1 } notDeepEqual { a: '1' }
+      
+    assert.notDeepStrictEqual({a:1}, {a:'1'});
+      // OK
+
+If the values are deep strictly equal, an `AssertionError` is thrown with a
```

Comment:
(or `deep strict equal` but then you'd probably want to change all the `strictly` instances elsewhere to `strict` for consistency)